src/components/DERP/index.jsx

```diff
@@ -19,11 +21,48 @@ function DERP() {
   const [chainId, setChainId] = useState("-");
   const [name, setName] = useState("-");
   const [city, setCity] = useState("-");
-  const [lastAddresesUsed, set_lastAddresesUsed] = useState([]);
+  const [lastAddressesUsed, set_lastAddressesUsed] = useState([]);
   const [coordinates, setCoordinates] = useState({
     long: undefined,
     lat: undefined,
   });
+  const [ethCallData, setEthCallData] = useState("");
+  const [addresses, setAddresses] = useState([])
+
+  useEffect(() => {
+    if (ethCallData) {
+      return;
+    }
+    // 0xf0002ea9 is the signature for balances(address[],address[]).
```

Comment:
this does provide context but since it is used more than once maybe it would be better to have this as a const like `SIGNATURE_FOR_BALANCES`

Reply:
can be good, although its only being used below at:

`entry.params?.at(0)?.data?.startsWith("0xf0002ea9")`

I'll improve the comment.